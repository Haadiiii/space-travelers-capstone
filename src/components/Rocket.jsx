import React from 'react';
import './rocket.css';

export default function Rocket(props) {
  return (
    <section className="rockets-section">
      <div className="">
        <img src={props.image} alt="rocket" />
      </div>
      <div className="details">
        <h1>{props.name}</h1>
        <p>{props.description}</p>
        <button className="button" type="button">Reserve Rocket</button>
      </div>
    </section>
  );
}
