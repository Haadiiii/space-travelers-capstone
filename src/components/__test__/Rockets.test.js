import React from "react";
import renderer from "react-test-renderer";
import Rocket from "./__mock__/rockect";

describe("A tests for my Rockets component", () => {
  it("Check whether Rockets component renders", () => {
    const rocket = {
      id: 1,
      image:
        "https://img.icons8.com/external-xnimrodx-blue-xnimrodx/64/000000/external-rocket-online-marketing-xnimrodx-blue-xnimrodx.png",
      name: "falcon1",
      description: "This is a description about Falcon 1",
      reserved: false,
    };
    const rocketTree = renderer
      .create(
        <Rocket
          key={rocket.id}
          image={rocket.image}
          description={rocket.description}
          name={rocket.name}
          reserved={rocket.reserved}
        />,
      )
      .toJSON();

    expect(rocketTree).toMatchSnapshot();
  });
  it('We now check whether it renders the rockets correctly', () => {
    const rockets = [];
    const rocketTree = renderer.create(
      <Rocket rockets={rockets} />,
    ).toJSON();

    expect(rocketTree).toMatchSnapshot();
  });
});
