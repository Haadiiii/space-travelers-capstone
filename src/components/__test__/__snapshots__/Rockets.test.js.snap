// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A tests for my Rockets component Check whether Rockets component renders 1`] = `
<section
  className="rockets-section"
>
  <div
    className=""
  >
    <img
      alt=""
      src=""
    />
  </div>
  <div
    className="details"
  >
    <h1 />
    <p />
    <button
      className="reserve button"
      onClick={[Function]}
      type="button"
    >
      Reserve Rocket
    </button>
  </div>
</section>
`;

exports[`A tests for my Rockets component We now check whether it renders the rockets correctly 1`] = `
<section
  className="rockets-section"
>
  <div
    className=""
  >
    <img
      alt=""
      src=""
    />
  </div>
  <div
    className="details"
  >
    <h1 />
    <p />
    <button
      className="reserve button"
      onClick={[Function]}
      type="button"
    >
      Reserve Rocket
    </button>
  </div>
</section>
`;
